sklearn/tree/tests/test_tree.py

```diff
@@ -2436,3 +2436,24 @@ def test_check_node_ndarray():
 
     with pytest.raises(ValueError, match="node array.+incompatible dtype"):
         _check_node_ndarray(problematic_node_ndarray, expected_dtype=expected_dtype)
+
+
+# TODO: Remove in v1.2
+@pytest.mark.parametrize("Tree", ALL_TREES.values())
+@pytest.mark.parametrize(
+    "old_max_features, new_max_features",
+    [
+        ("auto", "sqrt"),
+        ("auto", "sqrt"),
+    ],
+)
+def test_max_features_deprecated(Tree, old_max_features, new_max_features):
+    tree = Tree(max_features=old_max_features)
+
+    with pytest.warns(
+        FutureWarning, match="`max_features='auto'` has been deprecated in 1.1"
```

Comment:
We should test the different error messages for classifiers and regressors tho. The important part is the 

>  To keep the past behavior, explicitly set `max_features=`

which has a different message for classifiers vs regressors.